pkg/services/ngalert/provisioning/notification_policies.go

```diff
@@ -113,7 +115,11 @@ func (nps *NotificationPolicyService) UpdatePolicyTree(ctx context.Context, orgI
 
 	_, err = revision.Config.GetMergedAlertmanagerConfig()
 	if err != nil {
-		return definitions.Route{}, "", fmt.Errorf("new routing tree is not compatible with extra configuration: %w", err)
+		if errors.Is(err, definition.ErrSubtreeMatchersConflict) {
+			// TODO temporarily get the conflicting matchers
+			return definitions.Route{}, "", MakeErrRouteConflictingMatchers(fmt.Sprintf("%s", revision.Config.ExtraConfigs[0].MergeMatchers))
+		}
+		nps.log.Warn("Unable to validate the combined routing tree because of an error during merging. This could be a sign of broken external configuration. Skipping", "error", err)
```

Comment:
Do you think it's better to save than to return an error? It can be that the external config is broken, but it can also be that the merge is broken because of the new config for some reason. The saved config will be invalid and we won't be able to apply it later, so maybe return an error instead?

Reply:
Right, if the extra configurations are corrupted due to some reason, they won't be applied regardless of this API call is successful or not. Failing here in my opinion is exposing implementation details: user should not care about how we store the configuration.